server/app/services/program/ProgramServiceImpl.java

```diff
@@ -105,6 +106,18 @@ private CompletionStage<ProgramDefinition> syncProgramAssociations(Program progr
     return CompletableFuture.completedStage(programDefinition.orderBlockDefinitions());
   }
 
+  @Override
+  public ImmutableSet<String> getAllProgramNames() {
+    return programRepository.getAllProgramNames();
+  }
```

Comment:
This method isn't used in this PR but will be in a follow on for implementing the UI.